app/src/main/java/com/theopensourcefamily/chessclock/ClocksActivity.kt

```diff
@@ -36,12 +33,21 @@ class ClocksActivity : AppCompatActivity(), ClocksView {
     super.onDestroy()
     presenter.unbind()
   }
-  override fun render() {
+
+  override fun render(state: ClockState) {
     // Just to put something in the clocks right now
-    when(currentState) {
-      Stopped -> {}
-      WhiteRunning -> {}
-      BlackRunning -> {}
+    when (state) {
+      Stopped -> { }
+      WhiteRunning -> decreaseWhiteChrono()
+      BlackRunning -> decreaseBlackChrono()
     }
   }
+
+  private fun decreaseWhiteChrono() {
+    // get white current time (parsed in mm:ss) and decrease un second if it's greater than 00:00
```

Comment:
In order to avoid global parameters, I thought I could get text directly from whiteClock.text, parse it and decrease it one second... but It's not safe to do this because text is a String... What do you think about it? Do you have some better idea to avoid global parameter?

Reply:
Yess this will be done full reactively. Sorry about the little visibility I'm giving, but this logic would be nice to have in the presenter, and everything should be done in the RxStream. 
For the moment, while there is a State, usally that State depends on the previous State, and because of that a reducer function is usually applied. So for instance, if whites have 100 seconds, and the next state is also whites, the reducer would take 1 second from shites clock, and then return a new state (sometimes is called partial state and state).
So the reducer is just a function like this: `(State) -> State`